app/reactAdmin/components/Remote.jsx

```diff
@@ -32,9 +31,12 @@ class Remote extends React.Component {
   }
 
   tableComponents({ editable, columnList }) {
+    const { metadata } = this.props;
     const Component = (editable ? EditComponent : ShowComponent);
 
-    const columns = this.props.metadata[columnList ? columnList : editable ? "editColumns" : "showColumns"];
+    const columns = columnList
+      ? metadata[columnList]
+      : metadata[editable ? "editColumns" : "showColumns"];
```

Comment:
Why not the following? I feel like we're repeating the `metadata[]` for no good reason. The previous code had its line too long, but I think we can reindent it without changing the structure and its equally readable.

```
const columns = metadata[
  columnList
  ? columnList
  : editable ? "editColumns" : "showColumns"
];
```

Or maybe something like:

```
const columns = metadata[
   columnList ||
   (editable ? "editColumns" : "showColumns")
];
```

Reply:
Because eslint default rules suggests not nesting ternary conditions, which I agree. The second solution seems like the way to go here.